### DISCUSSION TITLE
멀티룸 zepetoplayers 설정 겹침 에러

### DISCUSSION THREAD

**Discussion by hipsc159** [naverz/zepeto-studio-kor#373] [5. Q&A]
가상의 평면을 만들고 OnTriggerEnter함수를 통해 지나가면 카메라의 줌거리 변경, 음악 변경을 하도록 했습니다.

혼자 플레이할때는 정상적으로 작동합니다.

하지만 멀티서버에서 여러명이 플레이할때, 문제가 발생합니다.

예를 들어, a,b가 있을 때, b가 지나가면 b의 설정만 변경되야 하지만 a, b 모두의 설정이 변경됩니다. 

유니티에서 여러 명이 접속했을 때 테스트를 못하니까 어떻게 해결해야할지 난감하네요.

이에 대해서 해결방법이 있을까요?

![스크린샷(318)](https://user-images.githubusercontent.com/68143858/153789874-7bc1f544-917e-483f-95d4-b9eaa07314bb.png)

![스크린샷(319)](https://user-images.githubusercontent.com/68143858/153790903-93424415-0e7f-4b75-a411-45a8b6a50790.png)

**Comment by x4cf** [comment]
로컬 캐릭터에 충돌채를 넣고, 충돌체로 채크하면 좀더 쉽게 구현할 수 있을거같아요

**Reply by hipsc159** [reply]
여기에 Collider를 넣으라는 말씀하시는 건가요?

![스크린샷(320)](https://user-images.githubusercontent.com/68143858/153823227-08298739-579d-497f-8173-ad53fe79d25f.png)

서버에 캐릭터가 접속할 때마다 캐릭터별로 구분할 수 있는 오브젝트가 있나요??

**Reply by x4cf** [reply]
사운드와 카메라에만 영향을 끼친다면 접속하는 모든 캐릭터를 구분할 필요는 없습니다. 자기자신의 캐릭터만 구분할 수 있으면 본인의 캐릭터가 충돌할때만 음악이나 카메라를 조작하도록 합니다.

**Reply by hipsc159** [reply]
아하 넵 알겠습니다. 
그럼 자신의 캐릭터만 구분할 수 있는 방법이 로컬 캐릭터라는 말씀으로 이해되는데, 
로컬 캐릭터에 충돌체를 넣고 충돌체를 체크하는게 어떤 건지 혹시 조금 더 구체적으로 알 수 있을까요?

**Reply by JeongYire** [reply]
자신에게만 영향을 주고싶다, 하시면 로컬캐릭터만 인식할 수 있게 코드를 짜시면 됩니다.
ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character << 이렇게 하시면 로컬캐릭터를 구분하실 수 있으니 
if문으로 onTriggerEnter를 통해 들어온 오브젝트와 , 위의 로컬캐릭터가 같은지 유무를 판단하게 하시면 될듯합니다 

ex) 

    OnTriggerEnter( data : GameObject ){
        if( ZepetoPlayers.instance.LocalPlayer != null ){
            if( ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.transform == data.transform ){
                /// 이 블록 부터는 로컬캐릭터만 접근가능 
            }
        }     
    }

**Reply by hipsc159** [reply]
헐 이제 확인하고 해봤는데, 됐어요... 정말 너무너무 감사합니다 ㅠㅠㅠ
